### PR TITLE
Add Get Submissions Endpoint

### PR DESCRIPTION
**Pull request by the4t4**
[GET] tasks/\<taskID>/submissions
query:
   userID=\<integer>
   submission=\<boolean>

If a user ID is specified in the query, then return the submission of that specific user.
If not, return all the submissions of the users assigned with that task.
Submission parameter decides if the solution is required to be returned or not.
The default behavior is false. Thus, only the grade and the user information are returned.

#78

### CODE REVIEW

File: backend/routes/tasks.js [hunk 1]
```diff
@@ -61,6 +62,45 @@ const exec = util.promisify(require('child_process').exec);
...
+
+    /* Get grade/s */
+    const result = await selectFromTable('grades', parameters);
+    if (result.error) 
+        return res.status(500).send(JSON.stringify({message: "Transaction Failed"}));
+    
+    const data = result.result.rows;
+
+    if (solutionEnable) {
+        const solutionRes = await selectFromTable('tasks', {taskID: tid})
```
Comment (Mohido) [inline backend/routes/tasks.js:92]: @the4t4, 
The solutions are found in the grades of the student as well. I think it is called `submission`. But I am not sure. The solution in the `tasks` table refers to the teacher's "perfect" solution, which the script will be comparing students' solutions with. For students, the solutions are found in the `grades` table.

**Review by Mohido — CHANGES_REQUESTED** [review]

File: backend/routes/tasks.js (hunk 1 above)
Comment (the4t4) [reply backend/routes/tasks.js:92]: You are confusing submission with solution, all the submissions are returned by default. In the issue you wrote to have a "solution" flag to return the solution as well.

Comment (Mohido) [reply backend/routes/tasks.js:92]: @the4t4 
Sorry for the confusion, what I meant is the submissions, not the solutions. The submissions should not be returned by default, they should be returned when the `solutions=true`, call it `submissions`, or anything that indicate the fact that the student solution will be returned with the query.

Comment (the4t4) [reply backend/routes/tasks.js]: Done

**Review by Mohido — APPROVED** [review]
LGMT

**Pull request merged**